modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -462,9 +462,17 @@ public void update(
 
 	private CPDefinition _addOrUpdateProduct(Product product) throws Exception {
 		CommerceCatalog commerceCatalog =
-			_commerceCatalogLocalService.getCommerceCatalog(
+			_commerceCatalogLocalService.fetchCommerceCatalog(
 				product.getCatalogId());
 
+		if (commerceCatalog == null) {
+			commerceCatalog =
+				_commerceCatalogLocalService.
```

Comment:
this could be null as well, and I think the scope of this ticket is to create the catalog via erc if it doesn't exist

Reply:
@victorcdp ignore this I think I've asked something wrong here. Sorry.